libs/application-generic/src/commands/project.command.ts

```diff
@@ -46,6 +47,11 @@ export abstract class EnvironmentWithUserCommand extends BaseCommand {
   readonly userId: string;
 }
 
+export abstract class EnvironmentWithUserObjectCommand extends BaseCommand {
+  @IsNotEmpty()
+  user: UserSessionData;
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** ‏ Is there a reason we need to pass all of the user data to the use case? i think we can reuse EnvironmentWithUserCommand or EnvironmentLevelWithUserCommand in the new use cases.

Reply:
I agree with your suggestion.

Reply:
we talked about it, @djabarovgeorge and I and reviewed it as a part of sync CR
For async prosperity, I've made it more clear I'm attempting to optimize tawards readability and maintainability, the existing object causing lots of overhead in all controllers as the controller resolves a session, this allows moving the session farwaerd, I don't see a reason to optimize less data availability ( as long as no additional latency / memory usage required.) I'm keeping it as is at the moment, it can be changed down the line if someone is have a very string opinion against making the full session object available for buisness logic